### PR TITLE
Userland: Implement a magnifier utility

### PR DESCRIPTION
**Pull request by vkoskiv**
Draft, since it still needs some work, namely: 

- [ ] Needs a 32x32 icon, I borrowed the 16x16 from the find button icon.
- [ ] ~MagnifierWidget should probably subclass ImageWidget instead of doing painting manually~
- [ ] ~Magnifier window should always be shown on the foreground~ (No facilities for this yet)
- [ ] Lacks double-buffering, tearing and flicker happens.
- [x] Bugs exist, and need to be fixed. (try dragging the magnifier window, and observe the global cursor tracking get out of sync)

I'd love feedback at this point while I add the finishing touches!

### CODE REVIEW

File: Userland/Applications/Magnifier/main.cpp [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        perror("pledge");
+        return 1;
+    }
+
+    auto app = GUI::Application::construct(argc, argv);
+
+    if (pledge("stdio cpath rpath recvfd sendfd", nullptr) < 0) {
+        perror("pledge");
+        return 1;
+    }
```
Comment (linusg) [inline Userland/Applications/Magnifier/main.cpp:27]: Let's also `unveil()` only the necessary paths :)

File: Userland/Applications/Magnifier/main.cpp [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+    // Sneaky!
+    // FIXME: Doesn't have a 32x32 icon yet, need to make one!
+    auto app_icon = GUI::Icon::default_icon("find");
+
+    auto window = GUI::Window::construct();
+    window->set_title("Magnifier");
+    window->resize(208, 208);
```
Comment (linusg) [inline Userland/Applications/Magnifier/main.cpp:35]: Magic numbers... probably 200 + whatever the 8px are? Would be nice to (`constexpr`) calculate this instead with some expressive names.

**Comment by vkoskiv** [comment]
This exposed a bug: `mouse_move` events are not sent to Magnifier while hovering over menus such as the Serenity menu, even though global tracking is enabled. I worked around this by just grabbing the global mouse position in my timer loop instead of the mouse move even handler.

File: Userland/Applications/Magnifier/MagnifierWidget.cpp [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    set_global_cursor_tracking(true);
+    GUI::WindowServerConnection::the().set_global_cursor_tracking(window_id, true);
+}
+
+void MagnifierWidget::set_scaling(int scale)
+{
+    VERIFY(scale == 2 || scale == 4);
+    m_scale_factor = scale;
+    update();
+}
```
Comment (linusg) [inline Userland/Applications/Magnifier/MagnifierWidget.cpp:37]: This feels a bit weird as it uses three different names for the same thing (`scale`, `scaling`, `scale_factor`). Suggestion:

```suggestion
void MagnifierWidget::set_scale_factor(int scale_factor)
{
    VERIFY(scale_factor == 2 || scale_factor == 4);
    m_scale_factor = scale_factor;
    update();
}
```

**Review by linusg — COMMENTED** [review]
commit message linter is failing :)

**Comment by vkoskiv** [comment]
I also added the `mag` alias, as suggested on discord.

**Comment by linusg** [comment]
Was nice watching you hack on this last night. Great work!

**Pull request merged**